### PR TITLE
Add Profiler Tab into Yii Debug Toolbar

### PR DESCRIPTION
**Pull request by tolidano**
The only thing missing from this amazing extension was the ability to profile each method call.  Using the sample provided by GoDaddy (from here: http://inside.godaddy.com/write-code-profiler-php/), we created a simple profiler and added the tab to the toolbar.  

Turning it on is as easy as this:
if ($dbg) { //whatever you use to decide if the toolbar should show up or not.
    require_once dirname(**FILE**).'/../extensions/yii-debug-toolbar/YiiDebugProfiler.php';
    declare(ticks=1);
    YiiDebugProfiler::start_profile();
}

### CODE REVIEW

File: yii-debug-toolbar/YiiDebugToolbarRoute.php [hunk 1]
```diff
@@ -109,7 +122,7 @@ public function init()
         parent::init();
 
         $this->enabled && $this->enabled = ($this->allowIp(Yii::app()->request->userHostAddress)
-                && !Yii::app()->getRequest()->getIsAjaxRequest() && (Yii::app() instanceof CWebApplication));
```
Comment (malyshev) [inline yii-debug-toolbar/YiiDebugToolbarRoute.php:112]: Should be run only for web application, excepts CLI and other

File: yii-debug-toolbar/YiiDebugToolbarRoute.php [hunk 2]
```diff
@@ -182,28 +195,9 @@ public function collectLogs($logger, $processLogs=false)
     protected function processLogs($logs)
     {
         $this->_endTime = microtime(true);
-        // disable log route based on white list
-        $this->enabled = $this->enabled && $this->checkContentTypeWhitelist();
         $this->enabled && $this->getToolbarWidget()->run();
     }
 
-    private function checkContentTypeWhitelist()
```
Comment (malyshev) [inline yii-debug-toolbar/YiiDebugToolbarRoute.php:190]: What is sense to remove this method?

File: yii-debug-toolbar/YiiDebugToolbarRoute.php [hunk 3]
```diff
@@ -20,21 +20,33 @@ class YiiDebugToolbarRoute extends CLogRoute
 
     private $_panels = array(
         'YiiDebugToolbarPanelServer',
-        'YiiDebugToolbarPanelRequest',
+        'YiiDebugToolbarPanelGlobals',
```
Comment (malyshev) [inline yii-debug-toolbar/YiiDebugToolbarRoute.php]: YiiDebugToolbarPanelGlobals? Where is it?

File: yii-debug-toolbar/panels/YiiDebugToolbarPanelProfiler.php [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+		}
+		return $total;
+	}
+
+    /**
+     * {@inheritdoc}
+     */
+    public function run()
+    {
+		YiiDebugProfiler::stop_profile();
```
Comment (malyshev) [inline yii-debug-toolbar/panels/YiiDebugToolbarPanelProfiler.php]: Use 4 white spaces for indents.

File: yii-debug-toolbar/panels/YiiDebugToolbarPanelProfiler.php [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+		$profile = YiiDebugProfiler::get_profile();
+		$totalTimes = $this->getTotalTimes($profile);
+		$finalProfiles = array();
+		foreach ($profile as $file=>$item) {
+			if (strpos($file, 'debug-toolbar')) {
```
Comment (malyshev) [inline yii-debug-toolbar/panels/YiiDebugToolbarPanelProfiler.php]: Put brackets to new line

File: yii-debug-toolbar/YiiDebugToolbarRoute.php (hunk 1 above)
Comment (tolidano) [inline yii-debug-toolbar/YiiDebugToolbarRoute.php:112]: Makes a lot of sense.

File: yii-debug-toolbar/YiiDebugToolbarRoute.php (hunk 2 above)
Comment (tolidano) [inline yii-debug-toolbar/YiiDebugToolbarRoute.php:190]: I used a slightly older copy of the file, all I really needed to do in this file was add the new Route at the top (where the other panels are listed)

File: yii-debug-toolbar/YiiDebugToolbarRoute.php (hunk 3 above)
Comment (tolidano) [inline yii-debug-toolbar/YiiDebugToolbarRoute.php]: It is called Globals in the version I had.  You may have renamed it at this point.

File: yii-debug-toolbar/panels/YiiDebugToolbarPanelProfiler.php (hunk 4 above)
Comment (tolidano) [inline yii-debug-toolbar/panels/YiiDebugToolbarPanelProfiler.php]: I'm a huge fan of 4-space real-tabs ( \t characters ) but I can certainly confirm.

File: yii-debug-toolbar/panels/YiiDebugToolbarPanelProfiler.php (hunk 5 above)
Comment (tolidano) [inline yii-debug-toolbar/panels/YiiDebugToolbarPanelProfiler.php]: Sure

**Comment by tolidano** [comment]
Let me make all these adjustments and be sure not to alter your Route improperly and send in a new request.

File: yii-debug-toolbar/panels/YiiDebugToolbarPanelProfiler.php (hunk 4 above)
Comment (tolidano) [inline yii-debug-toolbar/panels/YiiDebugToolbarPanelProfiler.php]: this file DOES use 4 white spaces - you're saying you want tabs instead, right?

Comment (malyshev) [inline yii-debug-toolbar/panels/YiiDebugToolbarPanelProfiler.php]: My comment not about file, but about line. Line 68 has no indent.

**Comment by tolidano** [comment]
Please review these again - I think we are much closer.

**Comment by malyshev** [comment]
Please rebase against master and squash all commits into a single commit.

**Comment by tolidano** [comment]
I believe I have done this in commit 76b63081c7febcad59e8f1cd61d22b2eda089448, but I am a little unclear though, and had to do a merge before pushing that squash to the origin.

**Comment by tolidano** [comment]
Did I do this correctly?  What can I do to correct?  Apologies, this is my first attempt at a pull request in GitHub.

Thanks a lot!

**Comment by tolidano** [comment]
old.

**Pull request closed without merging**